### PR TITLE
qa: Fix some tests to work on native windows

### PR DESCRIPTION
**Pull request by maflcko**
This allows some more tests to be run natively on Windows

### CODE REVIEW

File: test/functional/wallet_dump.py [hunk 1]
```diff
@@ -125,18 +129,18 @@ def run_test (self):
...
+            read_dump(wallet_enc_dump, addrs, script_addrs, hd_master_addr_unenc)
         assert_equal(found_addr, test_addr_count)
         assert_equal(found_script_addr, 2)
         assert_equal(found_addr_chg, 90*2 + 50)  # old reserve keys are marked as change now
         assert_equal(found_addr_rsv, 90*2) 
         assert_equal(witness_addr_ret, witness_addr)
 
         # Overwriting should fail
-        assert_raises_rpc_error(-8, "already exists", self.nodes[0].dumpwallet, tmpdir + "/node0/wallet.unencrypted.dump")
+        assert_raises_rpc_error(-8, "already exists", lambda: self.nodes[0].dumpwallet(wallet_enc_dump))
```
Comment (NicolasDorier) [inline test/functional/wallet_dump.py]: should be `wallet_unenc_dump`

Comment (MarcoFalke) [reply test/functional/wallet_dump.py]: Both exists, so it shouldn't matter when we are just testing that overwriting fails.

Also, the last calls to node 0 are to encrypt and then dump the encrypted wallet. So imo it is easier to understand the code (with less context) if we try to overwrite the dump of the encrypted wallet.

**Comment by MarcoFalke** [comment]
Should fix the following issues:

* mempool_persists:
```
FileExistsError: [WinError 183] Cannot create a file when that file already exists: 'C:\\Users\\win\\AppData\\Local\\Temp\\1\\bitcoin_test_runner_20180306_225759\\mempool_persist_612\\node0\\regtest\\mempool.dat' -> 'C:\\Users\\win\\AppData\\Local\\Temp\\1\\bitcoin_test_runner_20180306_225759\\mempool_persist_612\\node1\\regtest\\mempool.dat'
```
* feature_proxy:
```
ImportError: No module named 'fcntl'
```
* wallet_dump:
```
AssertionError: not(C:\Users\win\AppData\Local\Temp\1\bitcoin_test_runner_20180306_225759\wallet_dump_636/node0/wallet.unencrypted.dump == C:\Users\win\AppData\Local\Temp\1\bitcoin_test_runner_20180306_225759\wallet_dump_636\node0\wallet.unencrypted.dump)
```

**Comment by practicalswift** [comment]
utACK fa3528a85b05ea9507077f3eb340c9fb189251a6

**Comment by NicolasDorier** [comment]
utACK fa3528a85b05ea9507077f3eb340c9fb189251a6

**Pull request merged**